vault/expiration_test.go

```diff
@@ -3497,3 +3498,51 @@ func TestExpiration_listIrrevocableLeases_includeAll(t *testing.T) {
 		t.Errorf("bad lease count. expected %d, got %d", expectedNumLeases, numLeases)
 	}
 }
+
+func InduceDeadlock(t *testing.T, vaultcore *Core, expected uint32) {
```

Comment:
It's a little weird to have this in expiration_test.go, and have it depend on vaultcore.expiration, since there's nothing specifically about expiration in the test itself.